include/GoogleSync/GoogleSync.php

```diff
@@ -0,0 +1,1230 @@
+                $GLOBALS['log']->fatal(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Invalid AuthConfig JSON');
+                return false;
+            }
+            else {
+                return $authJson_local;
+            }
+        }
+    }
+
+    /**
+     * Add a user to the list of users to sync
+     *
+     * The user id is used as the key
+     *
+     * @param string $id : the SuiteCRM user id
+     * @param string $name : the SuiteCRM user name
+     *  Not really used for anything other than reference
+     *
+     * @return bool Success/Failure
+     */
+    public function addUser($id, $name)
+    {
+        if (array_key_exists($id, $this->users))
+        {
+            $GLOBALS['log']->warn(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . $id . ' already set');
+            return false;
+        }
+        else {
+            $this->users[$id] = $name;
+            $GLOBALS['log']->info(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . $id . ' set to ' . $this->users[$id]);
+            return true;
+        }
+    }
+
+    /**
+     * Remove a user from the list of users to sync
+     *
+     * @param string $id : the SuiteCRM user id
+     *
+     * @return bool Success/Failure
+     */
+    public function delUser($id)
+    {
+        if (!array_key_exists($id, $this->users))
+        {
+            $GLOBALS['log']->warn(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . $id . ' missing');
+            return false;
+        }
+        else {
+            unset($this->users[$id]);
+            return true;
+        }
+    }
+
+    /**
+     * Creates and Sets the Google client in the object
+     *
+     * @param string $id : the SuiteCRM user id
+     *
+     * @return bool Success/Failure
+     */
+    public function setClient($id)
+    {
+        if (!$gClient_local = $this->getClient($id))
+        {
+            return false;
+        }
+        else {
+            $this->gClient = $gClient_local;
+            return true;
+        }
+    }
+
+    /**
+     * Set the Google client up for the user by id
+     *
+     * @param string $id : the SuiteCRM user id
+     *
+     * @return Google_Client
+     */
+    public function getClient($id)
+    {
+        // Retrieve user bean
+        if (!isset($this->workingUser))
+        {
+            $this->workingUser = new User();
+        }
+
+        $this->workingUser->retrieve($id);
+
+        // Retrieve Access Token JSON from user preference
+        $accessToken = json_decode(base64_decode($this->workingUser->getPreference('GoogleApiToken', 'GoogleSync')), true);
+        if (!array_key_exists('access_token', $accessToken))
+        {
+            // The Token is invalid JSON or missing
+            $GLOBALS['log']->fatal(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Invalid or Missing AuthToken');
+            return false;
```

Comment:
Please update the PHP Doc Block. Return only has Google_Client. Please go through this file an update the doc blocks for methods which return multiple types

Reply:
Done.